cmd/goplugin-heroku/heroku/start.go

```diff
@@ -0,0 +1,26 @@
+package heroku
+
+import (
+	"github.com/lyraproj/lyra/cmd/goplugin-heroku/resource"
+	"github.com/lyraproj/pcore/pcore"
+	"github.com/lyraproj/pcore/px"
+	"github.com/lyraproj/servicesdk/grpc"
+	"github.com/lyraproj/servicesdk/service"
+)
+
+// Start the Heroku plugin
+func Start() {
+	pcore.Do(func(c px.Context) {
+		s := Server(c)
+		grpc.Serve(c, s)
+	})
+}
+
+func Server(c px.Context) *service.Server {
+	sb := service.NewServiceBuilder(c, "Heroku")
+
+	evs := sb.RegisterTypes("Heroku", resource.App{})
```

Comment:
The App type needs to denote that the AppID is a provided attribute and perhaps also that some other attributes are immutable (i.e. cannot be changed without removing and recreating the app).

Reply:
Is there any documentation on how to denote that said attribute is immutable?

Reply:
Not besides the example here: https://github.com/lyraproj/servicesdk/blob/a22cb98187cbe31cb4e5f90c7ad37835fca588ac/service/server_test.go#L313

It should show up as documentation for `RegisterTypes` if you're using a decent IDE (such as Goland).